backend/src/email/services/email.service.ts

```diff
@@ -110,30 +110,34 @@ const setCampaignCredential = (campaignId: number): Promise<[number, Campaign[]]
  * @param campaignId 
  */
 const getCampaignDetails = async (campaignId: number): Promise<GetCampaignDetailsOutput> => {
-  const campaignDetails: CampaignDetails =  (await Campaign.findOne({ 
-    where: { id: +campaignId }, 
-    attributes: [
-      'id', 'name', 'type', 'created_at', 'valid',
-      [literal('CASE WHEN "cred_name" IS NULL THEN False ELSE True END'), 'has_credential'],
-      [literal('s3_object -> \'filename\''), 'csv_filename'],
-    ],
-    include: [
-      {
-        model: JobQueue,
-        attributes: ['status', ['created_at', 'sent_at']],
-      },
-      {
-        model: EmailTemplate,
-        attributes: ['body', 'subject', 'params', 'reply_to'],
-      }],
-  }))?.get({ plain: true }) as CampaignDetails
-
-  const numRecipients: number = await EmailMessage.count(
-    {
-      where: { campaignId: +campaignId },
-    }
+  const [campaignDetails, numRecipients] =  await Promise.all(
+    [
+      Campaign.findOne({ 
+        where: { id: +campaignId }, 
+        attributes: [
+          'id', 'name', 'type', 'created_at', 'valid',
+          [literal('CASE WHEN "cred_name" IS NULL THEN False ELSE True END'), 'has_credential'],
+          [literal('s3_object -> \'filename\''), 'csv_filename'],
+        ],
+        include: [
+          {
+            model: JobQueue,
+            attributes: ['status', ['created_at', 'sent_at']],
+          },
+          {
+            model: EmailTemplate,
+            attributes: ['body', 'subject', 'params', 'reply_to'],
+          }],
+      }),
```

Comment:
why was `.get({ plain: true })` moved to the return statement?

Reply:
it has to be called on the result not the promise